wicket-core/src/test/java/org/apache/wicket/markup/html/form/AutoLabelTest.java

```diff
@@ -163,22 +167,80 @@ class MyTestPage2 extends TestPage
 
 	}
 
+	// WICKET-7101
+	@Test
+	void labelTagClassesAreUpdatedWhenRelatedFormComponentIsUpdated()
+	{
+		class MyAjaxTestPage extends TestPage
+		{
+			private static final long serialVersionUID = 1L;
+
+			// we define a label that is auto-synchronized with component
+			MyAjaxTestPage()
+			{
+				super("<label wicket:for='t' wicket:auto='true'><span class='label-text'>field</span></label>");

```

Comment:
This introduces a new attribute wicket:auto="true" that essentially say label is always repainted when component is re-rendered this way label is always up to date.

If not set label behaves as before. Thus, this change should be backward compatible.